### PR TITLE
Refactor depends on

### PR DESCRIPTION
**Pull request by baarkerlounger**
This refactors the existing depends on definition in the Form definition JSONs. Currently they all specify a method and a value to check for, this also adds the object class the method should be applied to, rather than assuming it's always the "case log".

The intention is that this allows us to also specify depends on conditions on the user - such as that a page should only show for a particular user role etc.

### CODE REVIEW

File: config/forms/2021_2022.json
Comment (dushan-madetech) [inline config/forms/2021_2022.json]: Do you want this in the refactor pr?

File: app/models/form.rb
Comment (dushan-madetech) [inline app/models/form.rb:172]: Could this blow up if object_instance is nil?

File: app/models/form.rb
Comment (dushan-madetech) [inline app/models/form.rb:167]: Could maybe shorten it to use a ternary operator I think? 
`object_instance = object == "case_log" ? case_log : nil`

Comment (baarkerlounger) [reply app/models/form.rb:167]: Sort of deliberate doing this to indicate where the next PR is going. i.e. "user" is an expected value that doesn't do anything yet but will.

File: app/models/form.rb
Comment (baarkerlounger) [reply app/models/form.rb:172]: Yes but that should never currently be the case (otherwise our existing code could also blow up)

File: config/forms/2021_2022.json
Comment (baarkerlounger) [reply config/forms/2021_2022.json]: Nope good catch

File: app/models/form.rb
Comment (dushan-madetech) [reply app/models/form.rb:172]: Fair enough, we could maybe add some logging here in the unlikely event it does blow up? Log the various values so we can easily debug if needed

Comment (dushan-madetech) [reply app/models/form.rb:172]: But it's not a strong opinion so happy to go with your call here

Comment (baarkerlounger) [reply app/models/form.rb:172]: Being changed in the next PR anyway but for the moment I think you'd absolutely want it to hard crash since it would indicate there was an error in our JSON and the form is likely to be wrong.

**Comment by dushan-madetech** [comment]
Just giving it a manual test before I give the approval 🙂

**Review by dushan-madetech — APPROVED** [review]
Seems to be working faultlessly, 🚢 it when you think it's ready

**Review by moarpheus — APPROVED** [review]

**Pull request merged**
